src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -97,12 +104,69 @@ class Results extends React.Component<Props, State> {
         loadOrganizationTags(api, organization.slug, selection);
       }
     }
+
+    if (prevState.confirmedQuery !== confirmedQuery) this.fetchTotalCount();
   }
 
+  canLoadEvents = async () => {
+    const {api, location, organization} = this.props;
+    const {eventView} = this.state;
+    let needConfirmation = false;
+    let confirmedQuery = true;
+    const currentQuery = eventView.getEventsAPIPayload(location);
+    const duration = eventView.getDays();
+
+    if (duration > 30 && currentQuery.project) {
+      let projectLength = currentQuery.project.length;
+
+      if (
+        projectLength === 0 ||
+        (projectLength === 1 && currentQuery.project[0] === '-1')
+      ) {
+        const results = await fetchProjectsCount(api, organization.slug);
```

Comment:
Should you catch this request failing? Right now the error will bubble up to the page level error boundary. You could catch it locally here and assume the next query is safe.